### DISCUSSION TITLE
Is it possible to make ripgrep respect a .gitignore (or any similar ignore file) even if a given directory is not a git repository?

### DISCUSSION THREAD

**Discussion by kartikynwa** [BurntSushi/ripgrep#2639] [Q&A] [answered]
It's a weird request so sorry about that. But wondering if this is possible. Would help me a lot. Something like this:

```
in ~
 ❯ mkdir test
mkdir: created directory 'test'

in ~
 ❯ cd test

in ~/test
 ❯ echo 'a.txt' > .gitignore

in ~/test
 ❯ echo "bonsai" > a.txt

in ~/test
 ❯ echo "bonsai" > b.txt

in ~/test
 ❯ rg bonsai
b.txt
1:bonsai
```

**Comment by BurntSushi** [comment, marked as answer]
Your example snippet doesn't seem to line up with the question. You're showing a snippet where ripgrep does respect the gitignore file despite it not being a git repository. Or are you showing what you want to have happen?

Have you read the guide? It discusses other ignore files such as .ignore and .rgignore. And there is also the `--no-require-git` flag.

**Reply by kartikynwa** [reply]
Got it. Thanks.

> Or are you showing what you want to have happen?

Yeah this was my intention.

But using .ignore is good enough for me. Thanks very much.
